pkg/asset/agent/agentconfig/agent_config.go

```diff
@@ -280,6 +284,28 @@ func (a *AgentConfig) validateAdditionalNTPSources(additionalNTPSourcesPath *fie
 	return allErrs
 }
 
+func (a *AgentConfig) validateRendevousIPNotWorker(rendezvousIP string, hosts []agent.Host) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if rendezvousIP != "" {
+		for i, host := range hosts {
+			hostPath := field.NewPath("Hosts").Index(i)
+			if strings.Contains(string(host.NetworkConfig.Raw), rendezvousIP) && host.Role != "master" {
+				if len(host.Role) > 0 {
+					errMsg := "Host " + host.Hostname + " is not of role 'master' and has the rendevousIP assigned to it. The rendevousIP must be assigned to a host of role 'master'"
+					allErrs = append(allErrs, field.Forbidden(hostPath.Child("Host"), errMsg))
+				} else {
+					errMsg := "Host " + host.Hostname + " has no role and has the rendevousIP assigned to it. The rendevousIP must be assigned to a host of role 'master'"
```

Comment:
Since the role name is optional this will break configurations where its not defined. For example, trying this in dev-scripts results in:
`level=fatal msg=failed to fetch Agent Installer ISO: failed to load asset "Agent Config": invalid Agent Config configuration: Hosts[0].Host: Forbidden: Host master-0 has no role and has the rendevousIP assigned to it. The rendevousIP must be assigned to a host of role 'master'`

Reply:
Yeah. I realize that it was unclear on what the team would like when the role is omitted. Is that a failure case for this validation? Does agent installer assign that IP to a master host?